src/Controls/tests/DeviceTests/Elements/Layout/LayoutTests.Android.cs

```diff
@@ -1,18 +1,61 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
 using System.Threading.Tasks;
+using Android.Graphics.Drawables;
+using AndroidX.AppCompat.Widget;
+using AndroidX.Core.Widget;
 using Microsoft.Maui.Controls;
-using Microsoft.Maui.Graphics;
+using Microsoft.Maui.DeviceTests.Stubs;
 using Microsoft.Maui.Handlers;
+using Microsoft.Maui.Hosting;
 using Microsoft.Maui.Platform;
 using Xunit;
 
 namespace Microsoft.Maui.DeviceTests
 {
 	public partial class LayoutTests
 	{
+		[Fact, Category(TestCategory.Layout)]
+		public async Task NestedButtonHasExpectedIconPosition()
```

Comment:
Can we update this test so that it takes in all four variations of `Button.ButtonContentLayout.ImagePosition` and then those all get validated as being in the correct place?

Reply:
Good idea. Updated. That caught a failure with Top as well - a failure which I oddly wasn't able to reproduce in the sandbox test app, but nevertheless it looks to be a legit failure, fixed by updating the Top logic similarly in ButtonExtensions, which I did.